### DISCUSSION TITLE
Are there xfreerdp command options related to terminating a established "session" w Windows?

### DISCUSSION THREAD

**Discussion by bmullan** [FreeRDP/FreeRDP#10246] [General]
Ubuntu 22.04 & Ubuntu 24.04

I use xfreerdp2 and 3 

The _xfreerdp command & Options are executed in a Bash script._

I can run windows 11 remoteapps ok from Ubuntu but I'm puzzled by one thing.

After launching the RemoteApp...
- using it, 
- then Terminating it

_The Remoteapp's Ubuntu session windows does go away_  **but** 
the Bash script itself **_does not_** terminate _w/o a ctrl-c_ 

Leading me to think that Windows11 session might still be active somehow.

So I executed the same xfreerdp command in a terminal w/out the Bash Script:

> $ xfreerdp3 /cert:ignore /u:$USER /dynamic-resolution /clipboard **_/app:program:'%windir%\explorer.exe'_** /v:x.x.x.x

_**The exact same thing happens!**   
I Click to close the Remote App "window", it disappears correctly BUT again
the xfreerdp3 command itself, also **does not** terminate back to the command prompt w/o using ctrl-c ?_

Leaving out the **_/app:program:'%windir%\explorer.exe'_** brings up the whole Windows Desktop ok.

Logging into Windows11, then later terminating it by just clicking on "X" to kill the app _the xfreerdp2/3 command **does** terminate normally._

I thought maybe xfreerdp itself had a command option  involved in this but I don't see anything obvious: 
> $ xfreerdp3 --help

**Comment by akallabeth** [comment]
no, that is only possible server side.
* login remotely and log out all your sessions in taskmanager or similar
* remoteapp terminates after the last application exited (although the server might take a lot of time to determine that, e.g. minutes)

**Reply by bmullan** [reply]
Thanks but I know those points...

What I tried to say in the Orig Post was that xfreerdp2 or 3 never exits whether run in a Bash script or on the CLI if using "remoteapp".

Since filing this issue I have continued to search and have _found 2 other "issues" that sound very much related to my problem:_

1. **[https://github.com/FreeRDP/FreeRDP/issues/9301](https://github.com/FreeRDP/FreeRDP/issues/9301)**
2. **[https://github.com/FreeRDP/FreeRDP/issues/7741](https://github.com/FreeRDP/FreeRDP/issues/7741)**

The #2 seems coincidental because when I execute xfreerdp for a RemoteApp, as I stated, xfreerdp doesn't exit when the RemoteApp exits and I have to CTRL-C to terminate it (or the script its running in).

And when I CTRL-C I do get one more message displayed:

> ***fuse:*** *writing device: Bad file descriptor*

\#2 Whose **TITLE** is...

 > *Killing xfreerdp* **leaves a hanging** ***FUSE*** **directory #7741** 

So to me,  this seems to be an **xfreerdp remoteapp** "bug" since all three of these cases were using RemoteApp

**Reply by akallabeth** [reply]
@bmullan no, `xfreerdp` exits as soon as the server tells the remote session has ended.
this takes some time, sometimes never.

there might be some magic to detect that condition before the server does, but we did not investigate that so far. 

the issues you reference are not related.
